src/parser/parser.yy

```diff
@@ -1797,9 +1797,15 @@ match_path_list
     }
 
 match_node
-    : L_PAREN match_alias R_PAREN {
-        $$ = new MatchNode(*$2, nullptr, nullptr);
-        delete $2;
+    : L_PAREN R_PAREN {
+        $$ = new MatchNode();
+    }
+    | parenthesized_expression {
+        auto& e = $1;
+        if (e->kind() != Expression::Kind::kLabel) {
+            throw nebula::GraphParser::syntax_error(@1, "Invalid node pattern");
```

Comment:
delete $1 before throw exception

Reply:
The expression memory is managed by ObjectPool, and there is no need for delete, which would result in double free.